diff --git a/workspace_tools/patch.py b/workspace_tools/patch.py
index da5ab99..fa05245 100644
--- a/workspace_tools/patch.py
+++ b/workspace_tools/patch.py
@@ -9,7 +9,7 @@ the first 8 locations in sector 0 of the flash. If the result is 0, then executi
 transferred to the user code.
 """
 from struct import unpack, pack
-
+import sys
 
 def patch(bin_path):
     with open(bin_path, 'r+b') as bin:
@@ -29,6 +29,9 @@ def is_patched(bin_path):
 
 
 if __name__ == '__main__':
-    bin_path = "C:/Users/emimon01/releases/emilmont/build/test/LPC1768/ARM/MBED_A1/basic.bin"
+    if len(sys.argv) != 2:
+        print "Usage: patch.py BINFILE"
+        exit(1)
+    bin_path = sys.argv[1]
     patch(bin_path)
     assert is_patched(bin_path), "The file is not patched"
